grant-project/grant-monorepo
# Proposal updates

Closes #101.

### What This Does

Implements a first pass at proposal updates. This includes:

* A new model and set of endpoints around proposal updates
* A form to create proposal updates, only visible to trustees
* UI to display a list of updates, and an individual full update when you click on one

### What Needs Improvement

* POST endpoint is un-authed
* No emails are sent to funders
* No direct links to updates
* No proposal comment threads
* No editing / deleting updates
* UI is pretty quick and dirty

### Steps to Test

Make sure you run db migrations before starting

* Create a proposal that you are a trustee of
* Go to updates, confirm it shows empty list
* Post an update, confirm it succeeds
* Go to the updates, confirm you see your update
* Click on the update, confirm you can read it in full
* Click on back, confirm you are returned to the list

## Screenshots

### Actions -> Post an Update

<img width="146" alt="screen shot 2018-11-01 at 11 34 34 am" src="https://user-images.githubusercontent.com/649992/47862167-e2065c00-ddca-11e8-831d-efea75431ed6.png">

### Update form

<img width="828" alt="screen shot 2018-11-01 at 11 34 28 am" src="https://user-images.githubusercontent.com/649992/47862168-e3d01f80-ddca-11e8-9ff2-f0e88db09066.png">

### Updates list

<img width="1228" alt="screen shot 2018-11-01 at 11 34 06 am" src="https://user-images.githubusercontent.com/649992/47862175-e894d380-ddca-11e8-83c7-1f4c178b6796.png">

### Full Update page

<img width="1228" alt="screen shot 2018-11-01 at 11 34 12 am" src="https://user-images.githubusercontent.com/649992/47862178-ea5e9700-ddca-11e8-8da9-db13a815a51d.png">
